x-pack/plugins/security_solution/public/app/deep_links/index.ts

```diff
@@ -316,18 +334,22 @@ const nestedDeepLinks: SecurityDeepLinks = {
 
 /**
  * A function that generates the plugin deepLinks
+ * @param enableExperimental ExperimentalFeatures arg
  * @param licenseType optional string for license level, if not provided basic is assumed.
+ * @param capabilities optional arg for app start capabilities
  */
 export function getDeepLinks(
+  enableExperimental: ExperimentalFeatures,
   licenseType?: LicenseType,
   capabilities?: ApplicationStart['capabilities']
 ): AppDeepLink[] {
   return topDeepLinks
     .filter(
       (deepLink) =>
-        deepLink.id !== SecurityPageName.case ||
-        capabilities == null ||
-        (deepLink.id === SecurityPageName.case && capabilities.siem.read_cases === true)
+        (deepLink.id !== SecurityPageName.case && deepLink.id !== SecurityPageName.ueba) ||
```

Comment:
What do you want to achieve here? Maybe I am missing something but I do not understand why cases are related to the ueba.

Reply:
I think this is clearer:

```
deepLink.id !== SecurityPageName.case || // Is not cases
deepLink.id !== SecurityPageName.ueba) || // Is not ueba
capabilities == null || // User does not hae privilages to Security Solution
(deepLink.id === SecurityPageName.case && capabilities.siem.read_cases === true) || // Is cases and the user has at least read access to cases
(deepLink.id === SecurityPageName.ueba && enableExperimental.uebaEnabled) // Is ueba and the flag is on.
```

Reply:
x-pack/plugins/security_solution/public/app/deep_links/index.ts:350:9 - error TS2367: This condition will always return 'true' since the types 'SecurityPageName.case' and 'SecurityPageName.ueba' have no overlap.

Reply:
Yes, you were right! I missed that 🙂 . Sorry!